account_payment_mode_auto_reconcile/models/account_invoice.py

```diff
@@ -82,6 +84,16 @@ def auto_reconcile_credits(self, partial_allowed=True):
                     continue
                 invoice.assign_outstanding_credit(credit.get('id'))
 
+    @api.multi
+    def _filter_payment_same_journal(self, credits):
+        """Keep only credits on the same journal than the invoice."""
+        self.ensure_one()
+        line_ids = [credit['id'] for credit in credits]
+        lines = self.env['account.move.line'].browse(line_ids).filtered(
+            lambda line: line.journal_id == self.journal_id
+        )
+        return [credit for credit in credits if credit['id'] in lines.ids]
```

Comment:
```suggestion
        return [credit for credit in credits_dict if credit['id'] in lines.ids]
```
:warning: Python constant: https://docs.python.org/3/library/constants.html#credits